src/__tests__/theme-provider.test.js

```diff
@@ -0,0 +1,108 @@
+/* eslint func-style:0 */
+import React, {Component} from 'react'
+import {render, mount} from 'enzyme'
+import * as jestGlamorReact from 'jest-glamor-react'
+import toJson from 'enzyme-to-json'
+// eslint-disable-next-line import/default
+import glamorous from '../'
+import ThemeProvider, {CHANNEL} from '../theme-provider'
+
+expect.extend(jestGlamorReact.matcher)
+expect.addSnapshotSerializer(jestGlamorReact.serializer)
+
+const getMockedContext = unsubscribe => ({
+  [CHANNEL]: {
+    getState: () => {},
+    setState: () => {},
+    subscribe: () => unsubscribe,
+  },
+})
+
+test('renders a component with theme', () => {
+  const Comp = glamorous.div(
+    {
+      color: 'red',
+    },
+    (props, theme) => ({padding: theme.padding}),
+  )
+  expect(
+    render(
+      <ThemeProvider theme={{padding: '10px'}}>
+        <Comp />
+      </ThemeProvider>,
+    ),
+  ).toMatchSnapshotWithGlamor()
+})
+
+test('theme properties updates get propagated down the tree', () => {
+  class Parent extends Component {
+    state = {
+      padding: '10px',
+    }
+    setPadding = () => this.setState({padding: '20px'})
```

Comment:
nit-pick, probably don't need this to be a public class field because you're calling it on the instance :) could probably just be a normal method.

Also, I think that enzyme exposes [a mechanism](https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/setState.md) to set state on an instance, so maybe we should just do it that way instead?